### PR TITLE
ARROW-5652: [CI] Fix lint docker image

### PR DESCRIPTION
**Pull request by fsaintjacques**

### CODE REVIEW

File: dev/lint/Dockerfile [hunk 1]
```diff
@@ -25,7 +25,11 @@ RUN apt-get update && \
       clang-tidy-7 \
       clang-tools-7
 
-RUN conda install flake8 && \
+COPY ci/conda_env_gandiva.yml \
+     /arrow/ci/
+
+RUN conda install -q -c conda-forge \
+    --file arrow/ci/conda_env_gandiva.yml flake8 && \
```
Comment (wesm) [inline dev/lint/Dockerfile]: This definitely reads weird (if I didn't know what was in conda_env_gandiva.yml I'd be scratching my head). Can we not get the linter dependencies from apt?

**Comment by wesm** [comment]
I think a better option is to build and test Gandiva in the base cpp/Dockerfile image. I'm going to make that change now

**Comment by wesm** [comment]
Done, though the tests fail running that Dockerfile:

```
est project /build/cpp
      Start   1: plasma-serialization_tests
      Start   2: plasma-client_tests
      Start   3: plasma-external_store_tests
      Start   4: arrow-allocator-test
 1/82 Test   #1: plasma-serialization_tests ................   Passed    0.06 sec
      Start   5: arrow-array-test
 2/82 Test   #4: arrow-allocator-test ......................   Passed    0.07 sec
      Start   6: arrow-buffer-test
 3/82 Test   #6: arrow-buffer-test .........................   Passed    0.04 sec
      Start   7: arrow-extension_type-test
 4/82 Test   #7: arrow-extension_type-test .................   Passed    0.04 sec
      Start   8: arrow-memory_pool-test
 5/82 Test   #8: arrow-memory_pool-test ....................   Passed    0.04 sec
      Start   9: arrow-pretty_print-test
 6/82 Test   #9: arrow-pretty_print-test ...................   Passed    0.04 sec
      Start  10: arrow-public-api-test
 7/82 Test  #10: arrow-public-api-test .....................   Passed    0.05 sec
      Start  11: arrow-scalar-test
 8/82 Test  #11: arrow-scalar-test .........................   Passed    0.04 sec
      Start  12: arrow-status-test
 9/82 Test  #12: arrow-status-test .........................   Passed    0.04 sec
      Start  13: arrow-stl-test
10/82 Test  #13: arrow-stl-test ............................   Passed    0.06 sec
      Start  14: arrow-type-test
11/82 Test  #14: arrow-type-test ...........................   Passed    0.05 sec
      Start  15: arrow-table-test
12/82 Test  #15: arrow-table-test ..........................   Passed    0.05 sec
      Start  16: arrow-table_builder-test
13/82 Test  #16: arrow-table_builder-test ..................   Passed    0.05 sec
      Start  17: arrow-tensor-test
14/82 Test  #17: arrow-tensor-test .........................   Passed    0.04 sec
      Start  18: arrow-sparse_tensor-test
15/82 Test   #3: plasma-external_store_tests ...............   Passed    0.65 sec
      Start  21: arrow-compute-test
16/82 Test  #18: arrow-sparse_tensor-test ..................   Passed    0.05 sec
      Start  22: arrow-compute-expression-test
17/82 Test  #21: arrow-compute-test ........................   Passed    0.05 sec
      Start  23: arrow-compute-operations-test
18/82 Test  #22: arrow-compute-expression-test .............   Passed    0.05 sec
      Start  25: arrow-compute-boolean-test
19/82 Test  #23: arrow-compute-operations-test .............   Passed    0.06 sec
      Start  26: arrow-compute-cast-test
20/82 Test  #25: arrow-compute-boolean-test ................   Passed    0.05 sec
      Start  27: arrow-compute-hash-test
21/82 Test  #26: arrow-compute-cast-test ...................   Passed    0.07 sec
      Start  28: arrow-compute-take-test
22/82 Test  #28: arrow-compute-take-test ...................   Passed    0.05 sec
      Start  29: arrow-compute-util-internal-test
23/82 Test  #29: arrow-compute-util-internal-test ..........   Passed    0.05 sec
      Start  30: arrow-compute-aggregate-test
24/82 Test  #27: arrow-compute-hash-test ...................   Passed    0.47 sec
      Start  32: arrow-compute-filter-test
25/82 Test   #5: arrow-array-test ..........................   Passed    1.67 sec
      Start  34: orc-adapter-test
26/82 Test  #34: orc-adapter-test ..........................   Passed    0.92 sec
      Start  35: arrow-feather-test
27/82 Test  #35: arrow-feather-test ........................   Passed    0.14 sec
      Start  36: arrow-ipc-read-write-test
28/82 Test  #32: arrow-compute-filter-test .................   Passed    1.92 sec
      Start  37: arrow-ipc-json-simple-test
29/82 Test  #37: arrow-ipc-json-simple-test ................   Passed    0.06 sec
      Start  38: arrow-ipc-json-test
30/82 Test  #38: arrow-ipc-json-test .......................   Passed    0.09 sec
      Start  39: arrow-json-integration-test
31/82 Test  #39: arrow-json-integration-test ...............   Passed    0.05 sec
      Start  41: arrow-concatenate-test
32/82 Test  #30: arrow-compute-aggregate-test ..............   Passed    2.44 sec
      Start  42: arrow-csv-chunker-test
33/82 Test  #42: arrow-csv-chunker-test ....................   Passed    0.05 sec
      Start  43: arrow-csv-column-builder-test
34/82 Test  #43: arrow-csv-column-builder-test .............   Passed    0.05 sec
      Start  44: arrow-csv-converter-test
35/82 Test  #41: arrow-concatenate-test ....................   Passed    0.11 sec
      Start  45: arrow-csv-parser-test
36/82 Test  #44: arrow-csv-converter-test ..................   Passed    0.06 sec
      Start  48: arrow-filesystem-test
37/82 Test  #45: arrow-csv-parser-test .....................   Passed    0.06 sec
      Start  49: arrow-localfs-test
38/82 Test  #48: arrow-filesystem-test .....................   Passed    0.06 sec
      Start  50: arrow-json-test
39/82 Test  #49: arrow-localfs-test ........................   Passed    0.06 sec
      Start  52: arrow-io-buffered-test
40/82 Test  #52: arrow-io-buffered-test ....................   Passed    0.06 sec
      Start  53: arrow-io-compressed-test
41/82 Test  #50: arrow-json-test ...........................   Passed    0.08 sec
      Start  54: arrow-io-file-test
42/82 Test  #54: arrow-io-file-test ........................   Passed    0.17 sec
      Start  55: arrow-io-memory-test
43/82 Test  #36: arrow-ipc-read-write-test .................   Passed    1.40 sec
      Start  56: arrow-io-readahead-test
44/82 Test  #56: arrow-io-readahead-test ...................   Passed    0.18 sec
      Start  59: arrow-utility-test
45/82 Test  #55: arrow-io-memory-test ......................   Passed    0.63 sec
      Start  60: arrow-bit-util-test
46/82 Test  #59: arrow-utility-test ........................   Passed    0.19 sec
      Start  61: arrow-compression-test
47/82 Test  #60: arrow-bit-util-test .......................   Passed    0.25 sec
      Start  62: arrow-decimal-test
48/82 Test  #61: arrow-compression-test ....................   Passed    0.21 sec
      Start  63: arrow-logging-test
49/82 Test  #62: arrow-decimal-test ........................   Passed    0.05 sec
      Start  64: arrow-rle-encoding-test
50/82 Test   #2: plasma-client_tests .......................   Passed    4.91 sec
      Start  65: arrow-task-group-test
51/82 Test  #63: arrow-logging-test ........................   Passed    0.18 sec
      Start  66: arrow-thread-pool-test
52/82 Test  #64: arrow-rle-encoding-test ...................   Passed    0.17 sec
      Start  67: arrow-uri-test
53/82 Test  #67: arrow-uri-test ............................   Passed    0.06 sec
      Start  79: arrow-dataset-file_test
54/82 Test  #79: arrow-dataset-file_test ...................***Failed    0.06 sec
Running arrow-dataset-file_test, redirecting output into /build/cpp/build/test-logs/arrow-dataset-file_test.txt (attempt 1/1)
/build/cpp/debug/arrow-dataset-file_test: error while loading shared libraries: libbrotlienc.so.1: cannot open shared object file: No such file or directory
/build/cpp/src/arrow/dataset

      Start  80: arrow-flight-test
55/82 Test  #80: arrow-flight-test .........................***Failed    0.05 sec
Running arrow-flight-test, redirecting output into /build/cpp/build/test-logs/arrow-flight-test.txt (attempt 1/1)
/build/cpp/debug/arrow-flight-test: error while loading shared libraries: libprotobuf.so.18: cannot open shared object file: No such file or directory
/build/cpp/src/arrow/flight

      Start  81: parquet-internals-test
56/82 Test  #65: arrow-task-group-test .....................   Passed    0.26 sec
      Start  82: parquet-reader-test
57/82 Test  #82: parquet-reader-test .......................   Passed    0.13 sec
      Start  83: parquet-writer-test
58/82 Test  #66: arrow-thread-pool-test ....................   Passed    0.44 sec
      Start  84: parquet-arrow-test
59/82 Test  #81: parquet-internals-test ....................   Passed    0.48 sec
      Start  85: parquet-file-deserialize-test
60/82 Test  #85: parquet-file-deserialize-test .............   Passed    0.07 sec
      Start  86: parquet-schema-test
61/82 Test  #86: parquet-schema-test .......................   Passed    0.05 sec
      Start  90: gandiva-internals-test
62/82 Test  #53: arrow-io-compressed-test ..................   Passed    2.16 sec
      Start  91: gandiva-precompiled-test
63/82 Test  #91: gandiva-precompiled-test ..................   Passed    0.03 sec
      Start  92: gandiva-filter_test
64/82 Test  #90: gandiva-internals-test ....................   Passed    0.17 sec
      Start  93: gandiva-projector_test
65/82 Test  #92: gandiva-filter_test .......................   Passed    0.23 sec
      Start  94: gandiva-projector_build_validation_test
66/82 Test  #94: gandiva-projector_build_validation_test ...   Passed    0.21 sec
      Start  95: gandiva-if_expr_test
67/82 Test  #95: gandiva-if_expr_test ......................   Passed    0.52 sec
      Start  96: gandiva-literal_test
68/82 Test  #96: gandiva-literal_test ......................   Passed    0.29 sec
      Start  97: gandiva-boolean_expr_test
69/82 Test  #97: gandiva-boolean_expr_test .................   Passed    0.50 sec
      Start  98: gandiva-binary_test
70/82 Test  #93: gandiva-projector_test ....................   Passed    1.82 sec
      Start  99: gandiva-date_time_test
71/82 Test  #98: gandiva-binary_test .......................   Passed    0.13 sec
      Start 100: gandiva-to_string_test
72/82 Test #100: gandiva-to_string_test ....................   Passed    0.06 sec
      Start 101: gandiva-utf8_test
73/82 Test #101: gandiva-utf8_test .........................   Passed    0.75 sec
      Start 102: gandiva-hash_test
74/82 Test  #99: gandiva-date_time_test ....................   Passed    1.01 sec
      Start 103: gandiva-in_expr_test
75/82 Test #102: gandiva-hash_test .........................   Passed    0.33 sec
      Start 104: gandiva-null_validity_test
76/82 Test #103: gandiva-in_expr_test ......................   Passed    0.18 sec
      Start 105: gandiva-decimal_test
77/82 Test #104: gandiva-null_validity_test ................   Passed    0.23 sec
      Start 106: gandiva-decimal_single_test
78/82 Test #105: gandiva-decimal_test ......................   Passed    1.01 sec
      Start 107: gandiva-projector_test_static
79/82 Test #107: gandiva-projector_test_static .............***Failed    0.06 sec
Running gandiva-projector_test_static, redirecting output into /build/cpp/build/test-logs/gandiva-projector_test_static.txt (attempt 1/1)
/build/cpp/debug/gandiva-projector_test_static: error while loading shared libraries: libbrotlienc.so.1: cannot open shared object file: No such file or directory
/build/cpp/src/gandiva/tests

80/82 Test  #84: parquet-arrow-test ........................   Passed    6.54 sec
81/82 Test #106: gandiva-decimal_single_test ...............   Passed    3.11 sec
82/82 Test  #83: parquet-writer-test .......................   Passed   13.39 sec

96% tests passed, 3 tests failed out of 82

Label Time Summary:
arrow-tests      =  15.74 sec*proc (53 tests)
arrow_dataset    =   0.06 sec*proc (1 test)
arrow_flight     =   0.05 sec*proc (1 test)
gandiva-tests    =  10.65 sec*proc (18 tests)
parquet-tests    =  20.66 sec*proc (6 tests)
plasma-tests     =   5.62 sec*proc (3 tests)
unittest         =  52.79 sec*proc (82 tests)

Total Test time (real) =  18.70 sec

The following tests FAILED:
	 79 - arrow-dataset-file_test (Failed)
	 80 - arrow-flight-test (Failed)
	107 - gandiva-projector_test_static (Failed)
Errors while running CTest
ninja: build stopped: subcommand failed.
```

This is obviously pretty concerning. But it is also failing in master so we'll have to investigate separately: https://travis-ci.org/ursa-labs/crossbow/builds/547691793#L2912

**Comment by fsaintjacques** [comment]
Already on it via https://jira.apache.org/jira/browse/ARROW-5653, the issue is `ninja install` and very likely linked to https://jira.apache.org/jira/browse/ARROW-5202.

**Comment by wesm** [comment]
Cool thanks :)

**Comment by wesm** [comment]
See https://stackoverflow.com/questions/32469953/why-is-cmake-designed-so-that-it-removes-runtime-path-when-installing

**Comment by wesm** [comment]
+1

**Comment by wesm** [comment]
@ursabot

**Comment by wesm** [comment]
@ursabot crossbow --help

**Comment by wesm** [comment]
@ursabot crossbow test --help

**Comment by wesm** [comment]
@ursabot crossbow test docker-iwyu

**Comment by wesm** [comment]
@ursabot crossbow test docker docker-iwyu

**Comment by wesm** [comment]
If it's possible to run an arbitrary docker task via ursabot that would be awesome, seems like it's not possible right now

**Pull request closed without merging**